bigframes/core/compile/single_column.py

```diff
@@ -179,4 +179,8 @@ def value_to_join_key(value: ibis_types.Value):
     """Converts nullable values to non-null string SQL will not match null keys together - but pandas does."""
     if not value.type().is_string():
         value = value.cast(ibis_dtypes.str)
-    return value.fillna(ibis_types.literal("$NULL_SENTINEL$"))
+    return (
+        value.fill_null(ibis_types.literal("$NULL_SENTINEL$"))
+        if hasattr(value, "fill_null")
+        else value.fillna(ibis_types.literal("$NULL_SENTINEL$"))
+    )
```

Comment:
fillna was renamed to fill_null in https://github.com/ibis-project/ibis/pull/9300 (ibis 9.1.0)

Reply:
Yes, `fillna` was renamed to `fill_null` in Ibis 9.1.0. Using `fillna` with this version or later will trigger a deprecation warning. However, we cann't use `fill_null` because it's not available in Ibis 9.0.0. Since Ibis 9.1.0 and later dropped support for Python 3.9, we must use Ibis 9.0.0 to maintain compatibility with Python 3.9

Reply:
Hmm, I do hate to have hasattr checks in our code. Should we just use fillna and suppress warnings?